Fix minor issues with Code Samples

mDuo13 opened the pull request:
- Remov a "feature" where the description gets deleted if it matches the title exactly (due to how our plugin finds the text of the README).
    | Before | After |
    |---|---|
    | ![price oracles description starts with '. '](https://github.com/user-attachments/assets/1c43a2b6-9068-42d0-8609-ab8ac4f573a7) | ![price oracles code sample has the correct description](https://github.com/user-attachments/assets/4e8a1bd6-4812-42ca-81aa-68f45e560df3) |
- Changed the top level of the Code Sampels layout to be a `<main>` element, which allows some styles to apply correctly and also helps the search crawler find the contents of the page.
    - The text now (correctly) does not turn purple when a card is hovered.
- Changed a couple code sample descriptions that had weird casing or typos.

mDuo13 committed:
Fix minor issues with Code Samples

- The description gets deleted if it matches the title exactly (due to
  how our plugin finds the text of the README). Changed a couple
  instances to not match exactly so this doesn't happen.
- Changed the top level of the Code Sampels layout to be a <main>
  element, which allows some styles to apply correctly and also helps
  the search crawler find the contents of the page.

amarantha-k commented:
The description on the last card needs to be fixed. 
<img width="510" alt="Screenshot 2025-01-29 at 4 18 01 PM" src="https://github.com/user-attachments/assets/dbac6495-482e-4b97-a722-68936a9c631f" />

mDuo13 committed:
Further fix code samples descriptions

mDuo13 commented:
Yep, that was another instance of the same problem. It turns out, the "feature" of removing text that matches the title was not even needed, so I removed it entirely. Also fixed another typo I spotted along the way.

AKCodez commented on resources/code-samples.page.tsx:
```diff
@@ -29,7 +29,7 @@ export default function CodeSamples() {
 
   return (
-    <div className="landing page-community">
+    <main className="landing page-code-samples">
```
I cant find any references to this className in the codebase. Maybe we can remove this? Otherwise LGTM

AKCodez replied:
<img width="373" alt="image" src="https://github.com/user-attachments/assets/2d66d4e7-165f-4c7c-a634-96009ae7c7d9" />

mDuo13 replied:
It's not used now but it's available if we need it later. Putting it there (on the correct element) is future-proofing.
